lib/smack/DSAWrapper.cpp

```diff
@@ -22,185 +24,94 @@ RegisterPass<DSAWrapper>
     DSAWrapperPass("dsa-wrapper",
                    "SMACK Data Structure Graph Based Alias Analysis Wrapper");
 
-void MemcpyCollector::visitMemCpyInst(llvm::MemCpyInst &mci) {
-  const llvm::EquivalenceClasses<const llvm::DSNode *> &eqs =
-      nodeEqs->getEquivalenceClasses();
-  const llvm::DSNode *n1 =
-      eqs.getLeaderValue(nodeEqs->getMemberForValue(mci.getOperand(0)));
-  const llvm::DSNode *n2 =
-      eqs.getLeaderValue(nodeEqs->getMemberForValue(mci.getOperand(1)));
-
-  bool f1 = false, f2 = false;
-  for (unsigned i = 0; i < memcpys.size() && (!f1 || !f2); i++) {
-    f1 = f1 || memcpys[i] == n1;
-    f2 = f2 || memcpys[i] == n2;
-  }
-
-  if (!f1)
-    memcpys.push_back(eqs.getLeaderValue(n1));
-  if (!f2)
-    memcpys.push_back(eqs.getLeaderValue(n2));
-}
-
 void DSAWrapper::getAnalysisUsage(llvm::AnalysisUsage &AU) const {
   AU.setPreservesAll();
-  AU.addRequiredTransitive<llvm::BUDataStructures>();
-  AU.addRequiredTransitive<llvm::TDDataStructures>();
-  AU.addRequiredTransitive<llvm::DSNodeEquivs>();
-  AU.addRequired<dsa::TypeSafety<llvm::TDDataStructures>>();
+  AU.addRequiredTransitive<sea_dsa::DsaAnalysis>();
 }
 
 bool DSAWrapper::runOnModule(llvm::Module &M) {
   dataLayout = &M.getDataLayout();
-  TD = &getAnalysis<llvm::TDDataStructures>();
-  BU = &getAnalysis<llvm::BUDataStructures>();
-  nodeEqs = &getAnalysis<llvm::DSNodeEquivs>();
-  TS = &getAnalysis<dsa::TypeSafety<llvm::TDDataStructures>>();
-  memcpys = collectMemcpys(M, new MemcpyCollector(nodeEqs));
-  staticInits = collectStaticInits(M);
-  module = &M;
-  return false;
-}
-
-std::vector<const llvm::DSNode *>
-DSAWrapper::collectMemcpys(llvm::Module &M, MemcpyCollector *mcc) {
-
-  for (llvm::Module::iterator func = M.begin(), e = M.end(); func != e;
-       ++func) {
-
-    for (llvm::Function::iterator block = func->begin(); block != func->end();
-         ++block) {
-
-      mcc->visit(*block);
+  SD = &getAnalysis<sea_dsa::DsaAnalysis>().getDsaAnalysis();
+  assert(SD->kind() == sea_dsa::GlobalAnalysisKind::CONTEXT_INSENSITIVE
+         && "Currently we only want the context-insensitive sea-dsa.");
+  for (auto& f : M.functions()) {
+    if(f.hasName() && SmackOptions::isEntryPoint(f.getName())) {
+      DG = &SD->getGraph(f);
```

Comment:
I don't get this. Why do we need this loop if there is just one big graph?

Reply:
I think sea-dsa's APIs only allow us to get the graph from a function. I may be able to get the graph by getting the first function in a module. Should I try it?

Reply:
Either that, or get it by looking for the entry point function (since we know that one is "normal" and it exists), and then when you find it, you get it from there. I would guess either way should work, and we should probably remove this loop.